test/new-e2e/tests/apm/tests.go

```diff
@@ -14,33 +14,32 @@ import (
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client/agentclient"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
 func testBasicTraces(c *assert.CollectT, service string, intake *components.FakeIntake, agent agentclient.Agent) {
 	traces, err := intake.Client().GetTraces()
-	require.NoError(c, err)
-	require.NotEmpty(c, traces)
-
-	trace := traces[0]
-	assert.Equal(c, agent.Hostname(), trace.HostName)
-	assert.Equal(c, "none", trace.Env)
-	require.NotEmpty(c, trace.TracerPayloads)
+	assert.NoError(c, err)
+	if assert.NotEmpty(c, traces) {
+		trace := traces[0]
+		assert.Equal(c, agent.Hostname(), trace.HostName)
+		assert.Equal(c, "none", trace.Env)
+		assert.NotEmpty(c, trace.TracerPayloads)
 
-	tp := trace.TracerPayloads[0]
-	assert.Equal(c, "go", tp.LanguageName)
-	require.NotEmpty(c, tp.Chunks)
-	require.NotEmpty(c, tp.Chunks[0].Spans)
-	spans := tp.Chunks[0].Spans
-	for _, sp := range spans {
-		assert.Equal(c, service, sp.Service)
-		assert.Contains(c, sp.Name, "tracegen")
-		assert.Contains(c, sp.Meta, "language")
-		assert.Equal(c, "go", sp.Meta["language"])
-		assert.Contains(c, sp.Metrics, "_sampling_priority_v1")
-		if sp.ParentID == 0 {
-			assert.Equal(c, float64(1), sp.Metrics["_dd.top_level"])
-			assert.Equal(c, float64(1), sp.Metrics["_top_level"])
+		tp := trace.TracerPayloads[0]
+		assert.Equal(c, "go", tp.LanguageName)
+		assert.NotEmpty(c, tp.Chunks)
```

Comment:
Not sure how much we care, but assert failures here cause a panic. 
The panics always seem to clutter the output. Maybe we can add an `if` and return early?

Not a big problem, just a personal annoyance with panicky tests.